src/test/llmtemplate_test.cpp

```diff
@@ -0,0 +1,332 @@
+    std::string jinjaFilePath;
+
+    void CreateConfig(std::string& fileContents, std::string& filePath) {
+        std::ofstream configFile{filePath};
+        SPDLOG_INFO("Creating config file: {}\n with content:\n{}", filePath, fileContents);
+        configFile << fileContents << std::endl;
+        configFile.close();
+        if (configFile.fail()) {
+            SPDLOG_INFO("Closing configFile failed");
+        } else {
+            SPDLOG_INFO("Closing configFile succeed");
+        }
+    }
+
+protected:
+    void SetUp() {
+        py::initialize_interpreter();
+        TestWithTempDir::SetUp();
+        tokenizerFilePath = directoryPath + "/tokenizer_config.json";
+        jinjaFilePath = directoryPath + "/template.jinja";
+    }
+    void TearDown() { py::finalize_interpreter(); }
+
+public:
+    void CreateTokenizerConfig(std::string& fileContents) {
+        CreateConfig(fileContents, tokenizerFilePath);
+    }
+    void CreateJinjaConfig(std::string& fileContents) {
+        CreateConfig(fileContents, jinjaFilePath);
+    }
+};
+
+TEST_F(LLMChatTemplateTest, ChatTemplateEmptyBody) {
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = "";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+    std::string errorOutput = "Expecting value: line 1 column 1 (char 0)";
+    ASSERT_EQ(finalPrompt, errorOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateEmptyMessage) {
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": []
+        }
+    )";
+    std::string errorOutput = "list object has no element 0";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+    ASSERT_EQ(finalPrompt, errorOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateSingleMessageError) {
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": {"role": "user", "content": "hello"}
+        }
+    )";
+    std::string errorOutput = "This servable accepts only single message requests";
```

Comment:
That's a bit odd. `messages` should be a list. I suppose in e2e flow it wouldn't pass earlier validation, so it's not really a valid case.

Reply:
I don't think it makes sense. Default template is not supposed to work with non-list messages. The output will be garbage anyway and we have one more test to maintain.